Cython/Distutils/build_ext.py

```diff
@@ -10,34 +11,126 @@
 else:
     from distutils.command.build_ext import build_ext as _build_ext
 
+try:
+    from __builtin__ import basestring
+except ImportError:
+    basestring = str
 
-class new_build_ext(_build_ext, object):
 
+class build_ext(_build_ext, object):
+
+    sep_by = _build_ext.sep_by
     user_options = _build_ext.user_options[:]
     boolean_options = _build_ext.boolean_options[:]
 
     user_options.extend([
+        ('cython-cplus', None,
+             "generate C++ source files"),
+        ('cython-create-listing', None,
+             "write errors to a listing file"),
+        ('cython-line-directives', None,
+             "emit source line directives"),
+        ('cython-include-dirs=', None,
+             "path to the Cython include files" + sep_by),
         ('cython-c-in-temp', None,
              "put generated C files in temp directory"),
+        ('cython-gen-pxi', None,
+            "generate .pxi file for public declarations"),
+        ('cython-directives=', None,
+            "compiler directive overrides"),
+        ('cython-gdb', None,
+             "generate debug information for cygdb"),
+        ('cython-compile-time-env', None,
+            "cython compile time environment"),
         ])
 
     boolean_options.extend([
-        'cython-c-in-temp'
+        'cython-cplus', 'cython-create-listing', 'cython-line-directives',
+        'cython-c-in-temp', 'cython-gdb',
     ])
 
     def initialize_options(self):
         _build_ext.initialize_options(self)
```

Comment:
I think we should generally call into `super(...)` here, rather than relying directly on the base class.

Reply:
fixed.